server/app/controllers/core.controller.js

```diff
@@ -22,73 +94,70 @@ exports.getAviDestinationMappings = asyncHandler(async (req, res, next) => {
-                                        'cloud_name': cloud_name,
-                                        seg: currentTenantSeg.map(item => {
-                                            return { 'name': item.name }
-                                        })
-                                    })
-                                }
+            const pythonProcess = spawn(DESTINATION_MAPPING_SCRIPT_PATH, [
+                '-c', avi_destination_ip,
+                '-u', avi_destination_user,
+                '-p', avi_destination_password,
+            ], {
+                shell: true,
+            });
+
+            let dataToSend;
+            let errorToSend;
+
+            // Collect data from script.
+            pythonProcess.stdout.on('data', function (data) {
+                console.log('Pipe data from python script ...');
+                dataToSend = data.toString();
+            });
+
+            pythonProcess.stderr.on('error', (error) => {
+                errorToSend = error.toString()
+                console.error(`Error while executing get_all_destination_names script, ${error}`);
+            });
+
+            pythonProcess.on('close', (code) => {
+                console.log(`child process close all stdio with code ${code}`);
+
+                if (code === 0 && fs.existsSync(DESTINATION_MAPPING_FILE)) {
+                    fs.readFile(DESTINATION_MAPPING_FILE, async (err, data) => {
+                        if (err) {
+                            console.error(`Error while reading ${DESTINATION_MAPPING_FILE}`, err);
+                            errorToSend = err;
+                        } else {
+                            // Implemented try catch to safely parse file data as JSON.
+                            let destinationFileData;
+                            let jsonParseError;
+
+                            try {
+                                destinationFileData = JSON.parse(data);
+                            } catch (err) {
+                                console.error(err);
+                                jsonParseError = err;
                             }
+
+                            if (destinationFileData) {
+                                parsedDestinationMappings = parseDestinationMappingsData(req, res, destinationFileData);
+                                // Insert parsedDestinationMappings to the DB.
+                                await AviDestinationMappingModel.insertMany({
+                                    avi_destination_ip,
+                                    tenants: parsedDestinationMappings.tenants
+                                });
+                                res.status(200).json(parsedDestinationMappings);
+                            } else {
+                                res.status(500).json({ error: `Error while parsing destination mapping response ${jsonParseError}` });
+                            }  
```

Comment:
We can improve error handling, can we not wait for fs function to complate and then do updating Db instead of doing in else block for that